### PR TITLE
Added type to consts

### PR DESCRIPTION
**Pull request by rsredsq**

### CODE REVIEW

File: Source/ToolCore/JSBind/JSBHaxe.cpp [hunk 1]
```diff
@@ -20,14 +20,8 @@ namespace ToolCore
...
-            if (ptype->kind_ == JSBPrimitiveType::Bool)
-                scriptType = "Bool";
-            if (ptype->kind_ == JSBPrimitiveType::Int && ptype->isUnsigned_)
-                scriptType = "UInt";
-            else if (ptype->kind_ == JSBPrimitiveType::Int)
-                scriptType = "Int";
-            if (ptype->kind_ == JSBPrimitiveType::Float)
-                scriptType = "Float";
+            scriptType = GetPrimitiveType(ptype);
+            return scriptType;
```
Comment (JoshEngebretson) [inline Source/ToolCore/JSBind/JSBHaxe.cpp]: Nice cleanup!

File: Source/ToolCore/JSBind/JSBHaxe.cpp [hunk 2]
```diff
@@ -61,6 +55,19 @@ namespace ToolCore
...
+        if (ptype->kind_ == JSBPrimitiveType::Bool)
+            return "Bool";
+        if (ptype->kind_ == JSBPrimitiveType::Int && ptype->isUnsigned_)
+            return "UInt";
+        else if (ptype->kind_ == JSBPrimitiveType::Int)
+            return "Int";
+        if (ptype->kind_ == JSBPrimitiveType::Float)
+            return "Float";
+        return "Dynamic";
+    }
```
Comment (JoshEngebretson) [inline Source/ToolCore/JSBind/JSBHaxe.cpp]: I like the dynamic primitive type :)  When do these happen?

File: Source/ToolCore/JSBind/JSBHeaderVisitor.h [hunk 3]
```diff
@@ -402,7 +403,14 @@ class JSBHeaderVisitor : public SymbolVisitor
...
         }
 
-        module_->RegisterConstant(getNameString(decl->name()).CString());
+        if (type->isIntegerType())
+        {
+            module_->RegisterConstant(getNameString(decl->name()).CString(), JSBPrimitiveType::Int);
+        }
+        else
+        {
+            module_->RegisterConstant(getNameString(decl->name()).CString(), JSBPrimitiveType::Float);
```
Comment (JoshEngebretson) [inline Source/ToolCore/JSBind/JSBHeaderVisitor.h]: Does this need to check for float or double?  Could it possibly be a string?  Maybe it should ignore if not a float/double?

File: Source/ToolCore/JSBind/JSBModule.cpp [hunk 4]
```diff
@@ -367,8 +368,7 @@ void JSBModule::RegisterConstant(const String& constantName)
         ErrorExit(ToString("Constant collision: %s", constantName.CString()));
     }
 
-    constants_.Push(constantName);
-
+    constants_[constantName] = new JSBPrimitiveType(type);
```
Comment (JoshEngebretson) [inline Source/ToolCore/JSBind/JSBModule.cpp]: Nice add on the typing!

**Comment by JoshEngebretson** [comment]
ok to test

**Comment by JoshEngebretson** [comment]
This looks great!  One question about float/double checking about... handing to CI!

**Comment by JoshEngebretson** [comment]
add to whitelist

File: Source/ToolCore/JSBind/JSBHaxe.cpp (hunk 2 above)
Comment (rsredsq) [inline Source/ToolCore/JSBind/JSBHaxe.cpp]: Dynamic shouldn't really happend, it's just to be safe :D

**Pull request merged**